user_tasks/signals.py

```
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+"""Celery signal handlers."""
+
+from __future__ import absolute_import, unicode_literals
+
+import logging
+from uuid import uuid4
+
+from django.contrib.auth import get_user_model
+from django.utils.module_loading import import_string
+
+from celery.signals import before_task_publish, task_failure, task_prerun, task_retry, task_success
+
+from .exceptions import TaskCanceledException
+from .models import UserTaskStatus
+from .tasks import UserTaskMixin
+
+LOGGER = logging.getLogger(__name__)
+
+
+@before_task_publish.connect
+def create_user_task(sender=None, body=None, **kwargs):  # pylint: disable=unused-argument
+    """
+    Create a :py:class:`UserTaskStatus` record for each :py:class:`UserTaskMixin`.
+
+    Also creates a :py:class:`UserTaskStatus` for each chain, chord, or group containing
+    the new :py:class:`UserTaskMixin`.
+    """
+    try:
+        task_class = import_string(sender)
+    except ImportError:
+        return
```

Why fail silently here on ImportError?